### PR TITLE
Redacting attribution Gateway access token

### PR DESCRIPTION
**Pull request by arafatkatze**
Solves https://github.com/sourcegraph/sourcegraph/issues/60045 

## Test plan
Run in admin and try to save access token and you shall get this. 
image

### CODE REVIEW

**Review by chwarwick — APPROVED** [review]
👍 looks like something that would be appropriate to backport

File: internal/conf/validate.go [hunk 1]
```diff
@@ -221,6 +221,7 @@ var siteConfigSecrets = []struct {
 	{readPath: `embeddings.accessToken`, editPaths: []string{"embeddings", "accessToken"}},
 	{readPath: `completions.accessToken`, editPaths: []string{"completions", "accessToken"}},
 	{readPath: `app.dotcomAuthToken`, editPaths: []string{"app", "dotcomAuthToken"}},
+	{readPath: `attribution\.gateway.accessToken`, editPaths: []string{"attribution.gateway", "accessToken"}},
```
Comment (keegancsmith) [inline internal/conf/validate.go:224]: @arafatkatze @chwarwick is this correct? I see you escape the `.` (`\.`). I don't see the other readPaths doing that, nor do you consistently do that in this string.

**Comment by keegancsmith** [comment]
@arafatkatze FYI to create a backport to the 5.3 branch you can just add the label `backport 5.3`. I see you manually did that, but gonna rely on automation here since it will create a more traceable commit message/etc.

File: internal/conf/validate.go (hunk 1 above)
Comment (arafatkatze) [reply internal/conf/validate.go:224]: This is correct. 

`	{readPath: `email\.smtp.username`, editPaths: []string{"email.smtp", "username"}},`

I copied it from here

Basically it supports
```

{
 "attribution.gateway": { 
    "accesstoken": "REDACTED"
   }
}
```

**Comment by arafatkatze** [comment]
@keegancsmith Thanks a lot for the help here and sorry for all the PR clutter.

**Pull request merged**